src/codegen/transformations.lisp

```diff
@@ -123,3 +124,37 @@ both CL namespaces appearing in `node`"
            (type parser:identifier-list bound-variables)
            (values boolean &optional))
   (null (intersection (node-variables node) bound-variables)))
+
+(defun node-returns-p (node)
+  "Does `node` contain a `node-return` subnode?"
+  (declare (type node node)
+           (values boolean &optional))
+  (traverse
+   node
+   (list
+    (action (:before node-return _node)
+      (declare (ignore _node))
+      (return-from node-returns-p t))))
+  nil)
+
+(defun localize-returns (node)
+  "If `node` contains any `node-return` subnodes, wrap `node` in a new
+`node-block` and rewrite all `node-return` subnodes to return from
+this block."
```

Comment:
Is there an established name for this?

Reply:
i would say this is about making returns explicit rather than localizing them.